tests/Samples.Server.Tests/BaseTest.cs

```diff
@@ -1,20 +1,29 @@
 ﻿using GraphQL.Samples.Server;
 using GraphQL.Server.Transports.AspNetCore.Common;
-using Microsoft.AspNetCore.TestHost;
+using Microsoft.AspNetCore.Mvc.Testing;
 using Newtonsoft.Json;
-using System;
 using System.Net.Http;
 using System.Text;
 using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Extensions.Hosting;
 
 namespace Samples.Server.Tests
 {
-    public abstract class BaseTest : IDisposable
+    public abstract class BaseTest : IClassFixture<WebApplicationFactory<Startup>>
     {
-        protected BaseTest()
+        private readonly WebApplicationFactory<Startup> _factory;
+
+        public BaseTest(WebApplicationFactory<Startup> factory)
         {
-            Server = new TestServer(Program.CreateWebHostBuilder(Array.Empty<string>()));
```

Comment:
Why not to stay `TestServer`? It is available in ASP.NET Core 3.0 and its api seems cleaner than `WebApplicationFactory<Startup>`. Although I suspect that something has changed there.

Reply:
Yea I tried staying with `TestServer` but the issue was that the call pattern changed. It still takes a `WebHostBuilder` even though in 3.0 your `Program` now creates a generic `HostBuilder`. I couldn't figure out the call pattern, so swapped over to the well-documented integration test approach which worked.

However, swapping over to the factory approach didn't work for the WebSockets test (see it's ctor) as it relies on the Startup class's assembly having a `Program` type with the `CreateHostBuilder` function (or `CreateWebHostBuilder` in 2.x) but those test/s are using a custom `Startup` straight in the test project.

I can swap this one back, but it's actually tidier like this (the new call pattern is quite long) and more standard/documented. I'd prefer to align the websockets one to this but obviously we can't. Also haven't got the WebSockets tests working yet, so...

Let me know.

Reply:
Does [this](https://github.com/aspnet/AspNetCore/issues/7754#issuecomment-543813345) not work?

Reply:
Yea it does, it's just more verbose and not documented. Want me to use that?

Reply:
Let's go this way. I would not say that the code is complicated. He is very straightforward.

Reply:
Another benefit of the factory is that it's the same code for both 2.x and 3.0, I'll have to dupe the blocks a bit to go back to the old way

Reply:
Well... I do not know. The testserver option seems simpler. Nevertheless the main requirement is that this just should work.

Reply:
Haha, I swapped over and it doesn't work. The response from the client is a 404... Honestly not keen on spending any more time trying to figure out why, though it could be the same issue affecting the websockets tests... hmmm.

Reply:
Nevermind, got it working. Alright, will consolidate the startup and program and then we'll be there almost.